build/lib/src/generate/run_builder.dart

```diff
@@ -24,7 +24,8 @@ Future<Null> runBuilder(Builder builder, Iterable<AssetId> inputs,
     AssetReader reader, AssetWriter writer, Resolvers resolvers,
     {Logger logger, String rootPackage}) async {
   logger ??= new Logger('runBuilder');
-  rootPackage ??= new Set.from(inputs.map((input) => input.package)).single;
+  rootPackage ??=
+      new Set<String>.from(inputs.map((input) => input.package)).single;
```

Comment:
the `as Iterable<String>` would fail with strong mode runtime semantics.

The `.toSet()` approach feels much better - good call!